File: 01_Language/01_Functions/python/is_callable.py
# coding: utf-8

def is_callable(var):
    if isinstance(var, str):
        try:
            return callable(globals()[var])
        except KeyError:
            return False

    return callable(var)


if __name__ == '__main__':
    def someFunction():
        pass


    functionVariable = "someFunction"
    print(is_callable(someFunction))
    print(is_callable(functionVariable))


    class someClass(object):
        def someMethod(self):
            pass


    anObject = someClass()
    print(is_callable(anObject.someMethod))

File: 01_Language/01_Functions/php/is_callable.php
<?php

//  How to check a variable to see if it can be called
//  as a function.

//
//  Simple variable containing a function
//

function someFunction()
{
}

$functionVariable = 'someFunction';

var_dump(is_callable($functionVariable, false, $callable_name));  // bool(true)

echo $callable_name, "\n";  // someFunction

//
//  Array containing a method
//

class someClass {

    function someMethod()
    {
    }

}

$anObject = new someClass();

$methodVariable = array($anObject, 'someMethod');

var_dump(is_callable($methodVariable, true, $callable_name));  //  bool(true)

echo $callable_name, "\n";  //  someClass::someMethod
